main/src/cgeo/geocaching/settings/Settings.java

```diff
@@ -175,7 +175,7 @@ private static void migrateSettings() {
             e.putInt(getKey(R.string.pref_version), prefsV0.getInt(getKey(R.string.pref_version), 0));
             e.putBoolean(getKey(R.string.pref_ratingwanted), prefsV0.getBoolean(getKey(R.string.pref_ratingwanted), true));
             e.putBoolean(getKey(R.string.pref_friendlogswanted), prefsV0.getBoolean(getKey(R.string.pref_friendlogswanted), true));
-            e.putBoolean(getKey(R.string.pref_useenglish), prefsV0.getBoolean(getKey(R.string.pref_useenglish), false));
+            e.putBoolean(getKey(R.string.old_pref_useenglish), prefsV0.getBoolean(getKey(R.string.old_pref_useenglish), false));
```

Comment:
As far as I have understood the code, this will not work as it will only be run if the settingsVersion is lower than 1. But on the other hand this feature isn't critical enough that it needs its own new settingsVersion...

Reply:
This is some very old migration code, not directly related to the current PR.

The selected language is stored in a new string preference called "selectedLanguage" (whereas the old "useenglish" was a boolean preference). "Migration" is done in `Settings.java`:

```
public static String getUserLanguage() {
    return getString(R.string.pref_selected_language, Settings.getBoolean(R.string.old_pref_useenglish, false) ? "en" : "");
}
```

Reply:
I always look at the code of others to learn something new, and this part had confused me a bit.Thanks for your explanation, now I got it and you're of course totally right!

Reply:
One addition: The change in the lines above came from the fact that I renamed the string which holds the actual preference key "useenglish" from "pref_useenglish" to "old_pref_useenglish" to mark it as outdated.